sdk/provisioning/README.md

```diff
@@ -0,0 +1,222 @@
+  }
+}
+```
+
+Because our infrastructure was configured to use interactive mode, the bicep
+is scoped to the resource group. This means that a resource group must be
+specified when deploying the bicep.
+
+### Using parameters
+
+It is possible to flow parameters to your bicep resources. Here is a simple
+example where a parameter is used to enable soft delete on the key vault.
+
+```C# Snippet:KeyVaultOnlyWithParameter
+var infrastructure = new SampleInfrastructure();
+
+var keyVault = infrastructure.AddKeyVault();
+
+keyVault.AssignProperty(
+    data => data.Properties.EnableSoftDelete,
+    new Parameter("enableSoftDelete", defaultValue: true, kind: BicepKind.Bool, description: "Enable soft delete for the key vault."));
+
+infrastructure.Build();
+```
+
+Here is the resulting bicep:
+
+```bicep
+targetScope = 'resourceGroup'
+
+@description('')
+param location string = resourceGroup().location
+
+@description('Enable soft delete for the key vault.')
+param enableSoftDelete bool = true
+
+
+resource keyVault_7LloDNJK5 'Microsoft.KeyVault/vaults@2022-07-01' = {
+  name: toLower(take('kv${uniqueString(resourceGroup().id)}', 24))
+  location: location
+  properties: {
+    tenantId: '00000000-0000-0000-0000-000000000000'
+    sku: {
+      family: 'A'
+      name: 'standard'
+    }
+    enableSoftDelete: enableSoftDelete
+    enableRbacAuthorization: true
+  }
+}
+```
+
+### Adding outputs
+
+It is possible to add outputs to your bicep. Here is a simple example where an output is added to the key vault.
+
+```C# Snippet:KeyVaultOnlyAddingOutput
+var infrastructure = new SampleInfrastructure();
+
+var keyVault = infrastructure.AddKeyVault();
+
+keyVault.AssignProperty(
+    data => data.Properties.EnableSoftDelete,
+    new Parameter("enableSoftDelete", defaultValue: true, kind: BicepKind.Bool, description: "Enable soft delete for the key vault."));
+
+keyVault.AddOutput("VAULT_URI", kind: BicepKind.String, data => data.Properties.VaultUri);
+
+infrastructure.Build();
+```
+
+Here is the resulting bicep:
+
+```bicep
+targetScope = 'resourceGroup'
+
+@description('')
+param location string = resourceGroup().location
+
+@description('Enable soft delete for the key vault.')
+param enableSoftDelete bool = true
+
+
+resource keyVault_7LloDNJK5 'Microsoft.KeyVault/vaults@2022-07-01' = {
+  name: toLower(take('kv${uniqueString(resourceGroup().id)}', 24))
+  location: location
+  properties: {
+    tenantId: '00000000-0000-0000-0000-000000000000'
+    sku: {
+      family: 'A'
+      name: 'standard'
+    }
+    enableSoftDelete: enableSoftDelete
+    enableRbacAuthorization: true
+  }
+}
+
+output VAULT_URI string = keyVault_7LloDNJK5.properties.vaultUri
+```
+
+## Troubleshooting
+
+-   File an issue via [GitHub Issues](https://github.com/Azure/azure-sdk-for-net/issues).
+-   Check [previous questions](https://stackoverflow.com/questions/tagged/azure+.net) or ask new ones on Stack Overflow using Azure and .NET tags.
+
+## Next steps
```

Comment:
Maybe link them to `azd` like https://learn.microsoft.com/azure/developer/azure-developer-cli/get-started here?